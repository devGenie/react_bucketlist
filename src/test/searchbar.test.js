import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, mount } from 'enzyme';
import sinon from 'sinon';
import { stub, spy } from 'sinon';
import 'should';
import 'should-enzyme';
import toJson from 'enzyme-to-json';
import sinonStubPromise from 'sinon-stub-promise';

sinonStubPromise(sinon)

import Search from '../components/search';
import bucketlistmock from '../__mocks__/bucketlists-mock';
import localStorageMock from '../__mocks__/localStorage';


window.sessionStorage = localStorageMock;

describe("<Search />",()=>{
	it("Should render",()=>{
		const wrapper=shallow(<Search/>);
		expect(toJson(wrapper)).toMatchSnapshot()
	})
	it("Should have an input field",()=>{
		const wrapper=shallow(<Search/>);
		let input = wrapper.find("input");
		expect(input).toHaveLength(1)
	});

	it("Should search",()=>{
		stub(window,'fetch').returnsPromise().resolves(bucketlistmock)
		const checkSearch = spy(Search.prototype,'search');
		const wrapper=shallow(<Search />);
		let input = wrapper.find("input");
		input.simulate("change",{
			target:{
				value:'search'
			}
		})
		expect(checkSearch.called).toBe(true)
		
	})
})