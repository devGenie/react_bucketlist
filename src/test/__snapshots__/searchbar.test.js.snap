// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Should render 1`] = `
<div
  className="searchbar z-depth-5 search"
>
  <input
    id="search"
    onChange={[Function]}
    placeholder="Search"
  />
</div>
`;
